app/src/main/java/com/lagradost/quicknovel/providers/MtlNovelsProvider.kt

```diff
@@ -0,0 +1,170 @@
+        Pair("Horror","horror"),
+        Pair("Josei","josei"),
+        Pair("Martial Arts","martial-arts"),
+        Pair("Mature","mature"),
+        Pair("Mecha","mecha"),
+        Pair("Military","military"),
+        Pair("Mystery","mystery"),
+        Pair("Psychological","psychological"),
+        Pair("Romance","romance"),
+        Pair("School Life","school-life"),
+        Pair("Sci-fi","sci-fi"),
+        Pair("Seinen","seinen"),
+        Pair("Shoujo","shoujo"),
+        Pair("Shoujo Ai","shoujo-ai"),
+        Pair("Shounen","shounen"),
+        Pair("Shounen Ai","shounen-ai"),
+        Pair("Slice of Life","slice-of-life"),
+        Pair("Smut","smut"),
+        Pair("Sports","sports"),
+        Pair("Supernatural","supernatural"),
+        Pair("Tragedy","tragedy"),
+        Pair("Two-dimensional","two-dimensional"),
+        Pair("Urban Life","urban-life"),
+        Pair("Wuxia","wuxia"),
+        Pair("Xianxia","xianxia"),
+        Pair("Xuanhuan","xuanhuan"),
+        Pair("Yaoi","yaoi"),
+        Pair("Yuri","yuri"),
+    )
+
+    override suspend fun loadMainPage(
+        page: Int,
+        mainCategory: String?,
+        orderBy: String?,
+        tag: String?
+    ): HeadMainPageResponse {
+        val url =
+            if (tag.isNullOrBlank()) "$mainUrl/alltime-rank/page/$page" else "$mainUrl/genre/$tag/page/$page"
+
+        val response = app.get(url)
+        val document = Jsoup.parse(response.text)
+
+        val headers =  document.select("div.box")
+        if (headers.size <= 0) return HeadMainPageResponse(url, ArrayList())
+        val returnValue: ArrayList<SearchResponse> = ArrayList()
+        for (h in headers) {
+            val name = h?.selectFirst("a")?.attr("aria-label")?.substringBeforeLast("Cover") ?: continue
+            val cUrl = h.selectFirst("a")?.attr("href") ?: throw ErrorLoadingException()
+            val posterUrl = h.selectFirst("amp-img amp-img")?.attr("src")
+            returnValue.add(
+                SearchResponse(
+                    name,
+                    cUrl,
+                    fixUrlNull(posterUrl),
+                    null,
+                    null,
+                    this.name
+                )
+            )
+        }
+        return HeadMainPageResponse(url, returnValue)
+    }
+
+    override suspend fun loadHtml(url: String): String? {
+        return app.get(url).document.selectFirst("div.par")?.html()
+    }
+
+
+    override suspend fun search(query: String): List<SearchResponse> {
+        val response =
+            SearchResults.fromJson(app.get("https://www.mtlnovel.com/wp-admin/admin-ajax.php?action=autosuggest&q=$query").text)
+        return response?.items?.first()?.results?.map { SearchResponse(Jsoup.parse(it.title).text(), it.permalink!!, it.thumbnail, null, null, name) }!! }
+
+
+    override suspend fun load(url: String): LoadResponse? {
+        val response = app.get(url)
+        val document = Jsoup.parse(response.text)
```

Comment:
you dont need to write Jsoup.parse everywhere, just write `val document = app.get(url).document`